packages/kbn-pm/src/commands/clean.ts

```diff
@@ -30,31 +30,47 @@ export const CleanCommand: ICommand = {
   description: 'Remove the node_modules and target directories from all projects.',
   name: 'clean',
 
-  async run(projects, projectGraph, { rootPath }) {
-    const directoriesToDelete = [];
+  async run(projects) {
+    const toDelete = [];
     for (const project of projects.values()) {
       if (await isDirectory(project.nodeModulesLocation)) {
-        directoriesToDelete.push(project.nodeModulesLocation);
+        toDelete.push({
+          cwd: project.path,
+          pattern: relative(project.path, project.nodeModulesLocation),
+        });
       }
 
       if (await isDirectory(project.targetLocation)) {
-        directoriesToDelete.push(project.targetLocation);
+        toDelete.push({
+          cwd: project.path,
+          pattern: relative(project.path, project.targetLocation),
+        });
       }
 
-      if (await isDirectory(project.optimizeLocation)) {
-        directoriesToDelete.push(project.optimizeLocation);
+      const { extraPatterns } = project.getCleanConfig();
+      if (extraPatterns) {
+        toDelete.push({
+          cwd: project.path,
+          pattern: extraPatterns,
+        });
       }
     }
 
-    if (directoriesToDelete.length === 0) {
-      log.write(chalk.bold.green('\n\nNo directories to delete'));
+    if (toDelete.length === 0) {
+      log.write(chalk.bold.green('\n\nNothing to delete'));
     } else {
-      log.write(chalk.bold.red('\n\nDeleting directories:\n'));
+      log.write(chalk.bold.red('\n\nDeleting:\n'));
 
-      for (const dir of directoriesToDelete) {
-        const deleting = del(dir, { force: true });
-        ora.promise(deleting, relative(rootPath, dir));
-        await deleting;
+      const originalCwd = process.cwd();
+      try {
+        for (const { pattern, cwd } of toDelete) {
+          process.chdir(cwd);
```

Comment:
__suggestion:__ can't we just pass `cwd` to `del` and get rid of all `process.chdir` related code?
```ts
for (const { pattern, cwd } of toDelete) {
    const promise = del(pattern, { cwd });
    ora.promise(promise, relative(process.cwd(), join(cwd, String(pattern))));
    await promise;
  }
```

Reply:
Del supports an undocumented cwd, but it doesn't apply to the current working directory "force" check, so I went with the chdir dance, but I'm open to `cwd` + `force`, though i'd like to avoid using `force`.

Reply:
Discussed offline and decided to stick with the chdir approach and add a comment to explain why.